### PR TITLE
Support using logical view in query aligned by time. 

### PR DESCRIPTION
**Pull request by citrusreticulata**
Support using logical view in query aligned by time. 

The commits do NOT support using logical view in query aligned by device till now.

### CODE REVIEW

File: server/src/main/java/org/apache/iotdb/db/mpp/plan/expression/visitor/ReplaceLogicalViewVisitor.java [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+ *
+ * <p>step 3. record paths that appeared in view expression. They should be fetched later.
+ */
+public class ReplaceLogicalViewVisitor extends ExpressionVisitor<Expression, List<PartialPath>> {
+
+  private TransformToExpressionVisitor transformToExpressionVisitor = null;
+
+  private GetSourcePathsVisitor getSourcePathsVisitor = null;
+
+  private boolean hadProcessedAggregationFunction = false;
```
Comment (MarcosZyk) [inline server/src/main/java/org/apache/iotdb/db/mpp/plan/expression/visitor/ReplaceLogicalViewVisitor.java:54]: rename to ```hasProcessedAggregationFunction``

File: server/src/main/java/org/apache/iotdb/db/mpp/plan/expression/visitor/ReplaceLogicalViewVisitor.java [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+
+  /** The paths that are new added, which should be re-fetched. */
+  private List<PartialPath> newAddedPathList = null;
```
Comment (MarcosZyk) [inline server/src/main/java/org/apache/iotdb/db/mpp/plan/expression/visitor/ReplaceLogicalViewVisitor.java:57]: rename to ```sourceTimeSeriesPathOfLogicalViewList``

**Review by MarcosZyk — COMMENTED** [review]
PTAL

**Comment by MarcosZyk** [comment]
Please add some IT.

File: server/src/main/java/org/apache/iotdb/db/mpp/plan/analyze/Analysis.java [hunk 3]
```diff
@@ -666,4 +666,27 @@ public Map<String, List<SortItem>> getDeviceToSortItems() {
...
+
+  public void setHasViewsInQuery(boolean hasViewsInQuery) {
+    this.hasViewsInQuery = hasViewsInQuery;
+  }
+
+  public boolean hasViewsInQuery() {
+    return this.hasViewsInQuery;
+  }
+
+  private List<Pair<Expression, String>> outputExpressions = null;
```
Comment (MarcosZyk) [inline server/src/main/java/org/apache/iotdb/db/mpp/plan/analyze/Analysis.java:683]: move the attribute to the upper place of this class to stay with other attributes.

File: server/src/main/java/org/apache/iotdb/db/mpp/plan/analyze/AnalyzeVisitor.java [hunk 4]
```diff
@@ -437,6 +443,69 @@ private void analyzeLastSource(
...
+        }
+      }
+    } catch (Exception e) {
+      throw new RuntimeException(e);
+    }
+    if (needToReFetch) {
+      ISchemaTree viewSchemaTree = this.schemaFetcher.fetchSchema(patternTree, null);
+      originSchemaTree.mergeSchemaTree(viewSchemaTree);
+    }
+    return originSchemaTree;
```
Comment (MarcosZyk) [inline server/src/main/java/org/apache/iotdb/db/mpp/plan/analyze/AnalyzeVisitor.java:473]: It is time-consuming to construct measurementPath. We can implement this check via a new method SchemaTree.hasLogicalView which can be get from the schema tree construction process.

File: server/src/main/java/org/apache/iotdb/db/mpp/plan/analyze/AnalyzeVisitor.java [hunk 5]
```diff
@@ -476,10 +551,11 @@ private Map<Integer, List<Pair<Expression, String>>> analyzeSelect(
...
                 ExpressionAnalyzer.removeAliasFromExpression(expression);
-            String alias =
-                !Objects.equals(expressionWithoutAlias, expression)
-                    ? expression.getExpressionString()
-                    : null;
+            //            String alias =
+            //                !Objects.equals(expressionWithoutAlias, expression)
+            //                    ? expression.getStringWithLogicalView()
+            //                    : null;
+            String alias = expression.getStringWithLogicalView();
```
Comment (MarcosZyk) [inline server/src/main/java/org/apache/iotdb/db/mpp/plan/analyze/AnalyzeVisitor.java:558]: remove the comments

File: server/src/main/java/org/apache/iotdb/db/mpp/plan/analyze/AnalyzeVisitor.java [hunk 6]
```diff
@@ -3033,30 +3157,84 @@ public Analysis visitCreateLogicalView(
...
+    for (Expression expression : sourceExpressionList) {
+      if (expression instanceof TimeSeriesOperand) {
+        pathsNeedCheck.add(((TimeSeriesOperand) expression).getPath());
       }
-      createLogicalViewStatement.setSourceExpressions(expressionList);
     }
-    analysis.setStatement(createLogicalViewStatement);
+    Pair<ISchemaTree, Integer> schemaOfNeedToCheck =
+        fetchSchemaOfPathsAndCount(pathsNeedCheck, context);
+    if (schemaOfNeedToCheck.right != pathsNeedCheck.size()) {
```
Comment (MarcosZyk) [inline server/src/main/java/org/apache/iotdb/db/mpp/plan/analyze/AnalyzeVisitor.java:3211]: This check seems not that efficient. Need improvement.

File: server/src/main/java/org/apache/iotdb/db/mpp/plan/expression/visitor/ReplaceLogicalViewVisitor.java (hunk 1 above)
Comment (citrusreticulata) [reply server/src/main/java/org/apache/iotdb/db/mpp/plan/expression/visitor/ReplaceLogicalViewVisitor.java]: fixed.

File: server/src/main/java/org/apache/iotdb/db/mpp/plan/expression/visitor/ReplaceLogicalViewVisitor.java (hunk 2 above)
Comment (citrusreticulata) [reply server/src/main/java/org/apache/iotdb/db/mpp/plan/expression/visitor/ReplaceLogicalViewVisitor.java:57]: fixed.

File: server/src/main/java/org/apache/iotdb/db/mpp/plan/analyze/Analysis.java (hunk 3 above)
Comment (citrusreticulata) [reply server/src/main/java/org/apache/iotdb/db/mpp/plan/analyze/Analysis.java]: fixed.

File: server/src/main/java/org/apache/iotdb/db/mpp/plan/analyze/AnalyzeVisitor.java (hunk 4 above)
Comment (citrusreticulata) [reply server/src/main/java/org/apache/iotdb/db/mpp/plan/analyze/AnalyzeVisitor.java]: Added a judgment, skip this step if it does not contain a view

**Review by liuminghui233 — APPROVED** [review]
LGTM :-)

**Pull request merged**